assets/src/components/ladder.tsx

```diff
@@ -128,21 +132,35 @@ const Ladder = ({
 const VehicleSvg = ({
   ladderVehicle,
   selectedVehicleId,
+  timepointStatusY,
 }: {
   ladderVehicle: LadderVehicle
   selectedVehicleId: VehicleId | undefined
+  timepointStatusY: (
+    timepointStatus: VehicleTimepointStatus | null,
+    direction: VehicleDirection
+  ) => number
 }) => {
   const dispatch = useContext(DispatchContext)
   const selectedClass =
     ladderVehicle.vehicle.id === selectedVehicleId ? "selected" : ""
 
+  const y =
+    isUnassignedBySwiftly(ladderVehicle) &&
+    ladderVehicle.vehicle.scheduledLocation
+      ? timepointStatusY(
+          ladderVehicle.vehicle.scheduledLocation.timepointStatus,
+          ladderVehicle.vehicleDirection
+        )
+      : ladderVehicle.y
```

Comment:
Yeah, this is good. I put all the properties we need into LadderVehicle that we need so that we could remove the vehicle. I also moved the `y` code all to one place. The component code is much simpler now which I think is a good sign.